egret/model_library/transmission/branch.py

```diff
@@ -276,20 +317,126 @@ def declare_eq_branch_power_dc_approx(model, index_set, branches, approximation_
         from_bus = branch['from_bus']
         to_bus = branch['to_bus']
 
-        x = branch['reactance']
         tau = 1.0
         shift = 0.0
-
         if branch['branch_type'] == 'transformer':
             tau = branch['transformer_tap_ratio']
             shift = math.radians(branch['transformer_phase_shift'])
 
-        b = 1/(tau*x)
+        if approximation_type == ApproximationType.BTHETA:
+            x = branch['reactance']
+            b = -1/(tau*x)
+        elif approximation_type == ApproximationType.BTHETA_LOSSES:
+            b = tx_calc.calculate_susceptance(branch)/tau
+
         m.eq_pf_branch[branch_name] = \
             m.pf[branch_name] == \
             b * (m.va[from_bus] - m.va[to_bus] - shift)
 
 
+def declare_eq_branch_loss_btheta_approx(model, index_set, branches, relaxation_type = RelaxationType.NONE):
+    """
+    Create the equality constraints for losses (from BTHETA approximation)
+    in the branch
+    """
+    m = model
+
+    con_set = decl.declare_set("_con_eq_branch_loss_btheta_approx_set", model, index_set)
+
+    m.eq_pfl_branch = pe.Constraint(con_set)
+    for branch_name in con_set:
+        branch = branches[branch_name]
+        if not branch['in_service']:
+            continue
```

Comment:
Comment I left on line 110 or so applies here.

Reply:
Thanks for the catch; we actually don't even need this if-case given my response to your prior comment!